source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -21,9 +21,11 @@ const std::vector<std::string>& defaultContentEncoding() {
                           "application/xhtml+xml"});
 }
 
-} // namespace
+std::vector<std::string>& registeredCompressors() {
+  MUTABLE_CONSTRUCT_ON_FIRST_USE(std::vector<std::string>);
```

Comment:
set?

Reply:
At the line 209 we are trying to adhere the principle of least surprise and get a compressor which was registered first. With sets or maps we'd loose this kind of ordering.